dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/ResetApproversActionlet.java

```diff
@@ -0,0 +1,130 @@
+
+package com.dotmarketing.portlets.workflows.actionlet;
+
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.workflows.model.WorkflowAction;
+import com.dotmarketing.portlets.workflows.model.WorkflowActionClassParameter;
+import com.dotmarketing.portlets.workflows.model.WorkflowActionFailureException;
+import com.dotmarketing.portlets.workflows.model.WorkflowActionletParameter;
+import com.dotmarketing.portlets.workflows.model.WorkflowHistory;
+import com.dotmarketing.portlets.workflows.model.WorkflowHistoryState;
+import com.dotmarketing.portlets.workflows.model.WorkflowHistoryType;
+import com.dotmarketing.portlets.workflows.model.WorkflowProcessor;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UUIDUtil;
+import com.dotmarketing.util.UtilMethods;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.dotmarketing.portlets.workflows.util.WorkflowActionletUtil.getParameterValue;
+
+/**
+ * Deletes the approvers of the workflow task
+ * @author jsanca
+ */
+public class ResetApproversActionlet extends WorkFlowActionlet {
+
+	private static final String ID_DELIMITER = ",";
+	private static final String PARAM_CONTENT_ACTIONS = "action";
+
+	private static ArrayList<WorkflowActionletParameter> ACTIONLET_PARAMETERS = null;
+
+	@Override
+	public synchronized List<WorkflowActionletParameter> getParameters() {
+		if (null == ACTIONLET_PARAMETERS) {
+			ACTIONLET_PARAMETERS = new ArrayList<>();
+			ACTIONLET_PARAMETERS
+					.add(new WorkflowActionletParameter(PARAM_CONTENT_ACTIONS,
+							"Optional Action ID, or Name", null,
+							false));
+		}
+		return ACTIONLET_PARAMETERS;
+	}
+
+	@Override
+	public String getName() {
+		return "Reset Approvals";
+	}
+
+	@Override
+	public String getHowTo() {
+
+		return "This actionlet will reset workflow history approvals";
+	}
+
+	private Optional<String> getWorkflowIdFromParameter (final WorkflowProcessor processor,
+														 final WorkflowActionClassParameter parameter)  {
+
+		final String value = getParameterValue(parameter);
+		if (null != value) {
+
+			if (!UUIDUtil.isUUID(value)) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4904110)